channels/tasks.py

```diff
@@ -131,3 +139,73 @@ def populate_subscriptions_and_roles(self):
         ]
     )
     raise self.replace(results)
+
+
+@app.task
+def populate_post_post_type(ids):
+    """
+    Populates Post.post_type for a range of posts
+
+    Args:
+        ids (list of int): list of post ids
+    """
+    client = get_admin_api()
+
+    for post in Post.objects.filter(id__in=ids).iterator():
+        if post.post_type:
+            continue
+
+        submission = client.get_submission(post.post_id)
+        if submission.url:
+            post.post_type = LINK_TYPE_LINK
+        elif hasattr(post, "article") and post.article:
+            post.post_type = EXTENDED_POST_TYPE_ARTICLE
+        else:
+            post.post_type = LINK_TYPE_SELF
+
+        post.save()
+
+
+@app.task
+def populate_channel_post_type(channel_ids):
+    """
+    Populates Channel.allowed_post_types for a range of channels
+
+    Args:
+        channel_ids (list of int): list channel post ids
+    """
+    client = get_admin_api()
+
+    for channel in Channel.objects.filter(id__in=channel_ids).iterator():
+        if channel.allowed_post_types:
+            continue
+
+        subreddit = client.get_subreddit(channel.name)
+        channel.allowed_post_types = allowed_post_types_bitmask(
+            get_allowed_post_types_from_link_type(subreddit.submission_type)
+        )
+        channel.save()
+
+
+@app.task(bind=True)
+def populate_post_and_channel_types(self):
+    """
+    Populates Channel.allowed_post_types and Post.post_type
+    """
+    results = celery.group(
+        [
+            populate_post_post_type.si(ids)
+            for ids in chunks(
+                Post.objects.values_list("id", flat=True),
+                chunk_size=settings.ELASTICSEARCH_INDEXING_CHUNK_SIZE,
+            )
+        ]
+        + [
+            populate_channel_post_type.si(ids)
```

Comment:
When run from `backpopulate_post_types` command, this prevents all existing channels from allowing creation of any new article posts (via `channels.api.get_allowed_post_types_from_link_type`).  Is this by design, so that channels have to be manually set to allow articles afterward?

Reply:
Yes, that's intentional, because we want to deploy this with no functional or configuration change to production.